modules/swagger-codegen/src/main/java/io/swagger/codegen/languages/ElixirClientCodegen.java

```diff
@@ -0,0 +1,377 @@
+
+        /**
+         * Template Location.  This is the location which templates will be read from.  The generator
+         * will use the resource stream to attempt to read the templates.
+         */
+        templateDir = "elixir";
+
+        /**
+         * Reserved words.  Override this with reserved words specific to your language
+         */
+        reservedWords = new HashSet<String>(
+                Arrays.asList(
+                        "sample1",  // replace with static values
+                        "sample2")
+        );
+
+        /**
+         * Additional Properties.  These values can be passed to the templates and
+         * are available in models, apis, and supporting files
+         */
+        additionalProperties.put("apiVersion", apiVersion);
+
+        /**
+         * Supporting Files.  You can write single files for the generator with the
+         * entire object tree available.  If the input file has a suffix of `.mustache
+         * it will be processed by the template engine.  Otherwise, it will be copied
+         */
+        supportingFiles.add(new SupportingFile("README.md.mustache",   // the input template or file
+                "",                                                       // the destination folder, relative `outputFolder`
+                "README.md")                                          // the output file
+        );
+        supportingFiles.add(new SupportingFile("config.exs.mustache",
+                "config",
+                "config.exs")
+        );
+        supportingFiles.add(new SupportingFile("mix.exs.mustache",
+                "",
+                "mix.exs")
+        );
+        supportingFiles.add(new SupportingFile("test_helper.exs.mustache",
+                "test",
+                "test_helper.exs")
+        );
+
+        /**
+         * Language Specific Primitives.  These types will not trigger imports by
+         * the client generator
+         */
+        languageSpecificPrimitives = new HashSet<String>(
+                Arrays.asList(
+                        "Type1",      // replace these with your types
+                        "Type2")
+        );
+    }
+
+    /**
+     * Configures the type of generator.
+     *
+     * @return the CodegenType for this generator
+     * @see io.swagger.codegen.CodegenType
+     */
+    public CodegenType getTag() {
+        return CodegenType.CLIENT;
+    }
+
+    /**
+     * Configures a friendly name for the generator.  This will be used by the generator
+     * to select the library with the -l flag.
+     *
+     * @return the friendly name for the generator
+     */
+    public String getName() {
+        return "elixir";
+    }
+
+    /**
+     * Returns human-friendly help for the generator.  Provide the consumer with help
+     * tips, parameters here
+     *
+     * @return A string value for the help message
+     */
+    public String getHelp() {
+        return "Generates an elixir client library.";
```

Comment:
@niku I suggest `Generates an elixir client library (alpha).` to indicate the API lib is still alpha.